nipype/interfaces/dipy/simulate.py

```diff
@@ -0,0 +1,274 @@
+
+        for f in self.inputs.in_dirs:
+            fd = nb.load(f).get_data()
+            args = np.hstack((args, fd[msk > 0]))
+
+        b0 = np.array([b0_im.get_data()[msk > 0]]).T
+        args = np.hstack((args, b0))
+
+        if isdefined(self.inputs.bval) and isdefined(self.inputs.bvec):
+            # Load the gradient strengths and directions
+            bvals = np.loadtxt(self.inputs.bval)
+            bvecs = np.loadtxt(self.inputs.bvec).T
+
+            # Place in Dipy's preferred format
+            gtab = gradient_table(bvals, bvecs)
+        else:
+            gtab = _generate_gradients(self.inputs.num_dirs,
+                                       self.inputs.bvalues)
+
+        np.savetxt(op.abspath(self.inputs.out_bvec), gtab.bvecs.T)
+        np.savetxt(op.abspath(self.inputs.out_bval), gtab.bvals.T)
+
+        snr = self.inputs.snr
+        args = [tuple(np.hstack((r, gtab, snr))) for r in args]
+
+        n_proc = self.inputs.n_proc
+        if n_proc == 0:
+            n_proc = cpu_count()
+
+        try:
+            pool = Pool(processes=n_proc, maxtasksperchild=50)
+        except TypeError:
+            pool = Pool(processes=n_proc)
+
+        iflogger.info(('Starting simulation of %d voxels, %d diffusion'
+                       ' directions.') % (len(args), len(gtab.bvals)))
+        result = pool.map(_compute_voxel, args)
+        ndirs = np.shape(result)[1]
+
+        simulated = np.zeros((shape[0], shape[1], shape[2], ndirs))
+        simulated[msk > 0] = result
+
+        simhdr = hdr.copy()
+        simhdr.set_data_dtype(np.float32)
+        simhdr.set_xyzt_units('mm', 'sec')
+        nb.Nifti1Image(simulated.astype(np.float32), aff,
+                       simhdr).to_filename(op.abspath(self.inputs.out_file))
+
+        return runtime
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        outputs['out_file'] = op.abspath(self.inputs.out_file)
+        outputs['out_mask'] = op.abspath(self.inputs.out_mask)
+        outputs['out_bvec'] = op.abspath(self.inputs.out_bvec)
+        outputs['out_bval'] = op.abspath(self.inputs.out_bval)
+
+        return outputs
+
+
+def _compute_voxel(args):
+    """
+    Simulate DW signal for one voxel. Uses the multi-tensor model and
+    three isotropic compartments.
+
+    Apparent diffusivity tensors are taken from [Alexander2002]_
+    and [Pierpaoli1996]_.
+
+    .. [Alexander2002] Alexander et al., Detection and modeling of non-Gaussian
+      apparent diffusion coefficient profiles in human brain data, MRM
+      48(2):331-340, 2002, doi: `10.1002/mrm.10209
+      <http://dx.doi.org/10.1002/mrm.10209>`_.
+    .. [Pierpaoli1996] Pierpaoli et al., Diffusion tensor MR imaging
+      of the human brain, Radiology 201:637-648. 1996.
+    """
+    D_ball = [3000e-6, 960e-6, 680e-6]
+    sf_evals = [1700e-6, 200e-6, 200e-6]
+
+    vfs = [args[0], args[1], args[2]]
+    ffs = [args[3], args[4], args[5]]  # single fiber fractions
```

Comment:
Always exactly three? Why not take lists as arguments and allow users to determine how many fibers they want?